docs/jsdoc.rst

```diff
@@ -0,0 +1,160 @@
+
+    /\*\*\@namespace firefly\*/
+
+    /\*\*\@namespace firefly.action\*/
+
+    /\*\*\@namespace firefly.ui\*/
+
+@param
+
+    This tag requires you to specify the name of the parameter you are
+    documenting. You can also include the parameter's type, enclosed in
+    curly brackets, and a description of the parameter. You can use \|
+    to specify more than one type such as {type1\|type2\|type3}.
+
+@static
+
+    This tag indicates that a symbol is contained within a parent and
+    can be accessed without instantiating the parent.
+
+@summary
+
+    This tag is a shorter version of the full description. It can be
+    added to any doclet.
+
+@typedef [<type>] <namepath>
+
+    This tag is useful for documenting custom types, particularly if you
+    wish to refer to them repeatedly. These types can then be used
+    within other tags expecting a type, such
+    as \@type or \@param.
+
+    NOTE: I used @global in each @typedef object. Thus, the typedef
+    object can be referred throughout the documentation.
+
+**When to use @func**
+
+By default, the jsdoc documents all the exported members by
+export.functionName. To document them as static members of a namespace,
+you use @func <functionName>. For methods and constants are defined
+without export, you don’t need to use @func <functionName>. **Note: The
+behavior in namespace is opposite to module.**
+
+**When to use @memberof**
+
+For methods or constants, you have to use @memberof <namespace> to
+group them under the <namespace>. Without @memberof tag, all the methods
+and constants will appear under the global category.
+
+**Known issues:**
+
+1. There is more than one way to produce the same output using the tags.
+
+2. Depending on the templates, the tags may not work exactly as they are
+   described. The default template came with JSDoc is located in the
+   jsdoc directory
+
+3. Note: the following two tags worked for minami template. But the
+   docstrap template seems not recognizing them correctly.
+
+    @private
+
+    The @private tag marks a symbol as private, or not meant for general
+    use. Private members are not shown in the generated output unless
+    JSDoc is run with the -a/--private command-line option or specifying
+    the option in the configuration. The @private tag is equivalent
+    to @access private.
+
+    @public
+
+    The @public tag indicates that a symbol should be documented as if
+    it were public. By default, JSDoc treats all symbols as public, so
+    using this tag does not normally affect the generated documentation.
+    However, you may prefer to use the @public tag explicitly so it is
+    clear to others that you intended to make the symbol public.
+    However, @public does not change the symbol’s scope. The @public tag
+    is the same as @access public.
+
+**Start from scratch:**
+
+1. Add the dependent libraries to package.json
+
+    "eslint-plugin-jsx-a11y": "^0.6.2",
```

Comment:
This could probably go in a code-block directive,

```rst
.. code-block:: json

   "eslint-plugin-jsx-a11y": "^0.6.2"
   "jsdoc-jsx": "^0.1.0",
   "ink-docstrap: "^1.2.1"
```